File: Ejercicios/Java/Interfaces/ConsolaLog.java
interface ILog {
    void log(String mensaje);
}

public class ConsolaLog implements ILog {
    private String prefijo;

    public ConsolaLog() {
        this("[INFO]");
    }

    public ConsolaLog(String prefijo) {
        this.prefijo = prefijo;
    }

    public ConsolaLog(String prefijo, boolean mayusculas) {
        this.prefijo = mayusculas ? prefijo.toUpperCase() : prefijo;
    }

    public ConsolaLog(ConsolaLog otro) {
        this.prefijo = otro.prefijo;
    }

    @Override
    public void log(String mensaje) {
        System.out.println(prefijo + " " + mensaje);
    }
}

File: Ejercicios/JavaScript/Interfaces/ConsolaLog.js
class ConsolaLog {
  constructor(prefijo, mayusculas) {
    if (prefijo === undefined && mayusculas === undefined) {
      this.prefijo = "[INFO]";
    } else if (prefijo instanceof ConsolaLog) {
      this.prefijo = prefijo.prefijo;
    } else if (mayusculas === undefined) {
      this.prefijo = prefijo;
    } else {
      this.prefijo = mayusculas ? prefijo.toUpperCase() : prefijo;
    }
  }

  log(mensaje) {
    console.log(this.prefijo, mensaje);
  }
}

export default ConsolaLog;
